controllers/usersignup/usersignup_controller.go

```diff
@@ -756,6 +765,71 @@ func (r *Reconciler) ensureSpaceBinding(
 	return nil
 }
 
+// annotateCaptchaAssessment attempts to annotate the original assessment (if any) to provide feedback so that future scores can be improved
+// the attempt to annotate the previous assessment is a best-effort, we do not want to reconcile again if the request fails because captcha is a separate service
+// and should not have any impact to the UserSignup flows if there are issues with this request
+func (r *Reconciler) annotateCaptchaAssessment(ctx context.Context, userSignup *toolchainv1alpha1.UserSignup, newState string) {
+	logger := log.FromContext(ctx)
+	assessmentID, assessmentIDFound := userSignup.Annotations[toolchainv1alpha1.UserSignupCaptchaAssessmentIDAnnotationKey]
+	if !assessmentIDFound {
+		// there's no assessment ID to annotate
+		return
+	}
+
+	newAssessmentAnnotation := r.shouldAnnotateCaptchaAssessment(userSignup, newState)
+	if newAssessmentAnnotation == recaptchapb.AnnotateAssessmentRequest_ANNOTATION_UNSPECIFIED {
+		// no need to annotate the previous assessment
+		return
+	}
+
+	gctx := context.Background()
+	rclient, err := recaptcha.NewClient(gctx)
+	if err != nil {
+		logger.Error(err, "error creating reCAPTCHA client, cannot annotate assessment")
+		return
+	}
+	defer rclient.Close()
+
+	annotateRequest := &recaptchapb.AnnotateAssessmentRequest{
+		Name:       assessmentID,
+		Annotation: newAssessmentAnnotation,
+	}
+
+	response, err := rclient.AnnotateAssessment(gctx, annotateRequest)
```

Comment:
I guess we could do it a go routine so we don't wait for the recapcha client here?

Reply:
Good idea 3e1681db3961ab2e855896d306a2a6e9d30b93fd